skeletons/INTEGER.h

```diff
@@ -43,6 +43,8 @@ xer_type_encoder_f INTEGER_encode_xer;
 per_type_decoder_f INTEGER_decode_uper;
 per_type_encoder_f INTEGER_encode_uper;
 
+#define INTEGER_constraint asn_generic_no_constraint
```

Comment:
Of course your repository do not need this line, as lone as you have apply #143 to it.

Please please take a look at the generated Version.c using your repository and also the Version.c generated by my #143.

You can see the Version.c generated by your repository

```
int
Version_constraint(asn_TYPE_descriptor_t *td, const void *sptr,
			asn_app_constraint_failed_f *ctfailcb, void *app_key) {
	/* Replace with underlying type checker */
	td->check_constraints = asn_DEF_INTEGER.check_constraints;
	return td->check_constraints(td, sptr, ctfailcb, app_key);
}
......
asn_TYPE_descriptor_t asn_DEF_Version = {
	"Version",
	"Version",
	Version_free,
	Version_print,
	Version_constraint,
	Version_decode_ber,
	Version_encode_der,
	Version_decode_xer,
	Version_encode_xer,
	0, 0,	/* No PER support, use "-gen-PER" to enable */
	0, 0,	/* No APER */
	0,	/* Use generic outmost tag fetcher */
	asn_DEF_Version_tags_1,
	sizeof(asn_DEF_Version_tags_1)
		/sizeof(asn_DEF_Version_tags_1[0]), /* 1 */
	asn_DEF_Version_tags_1,	/* Same as above */
	sizeof(asn_DEF_Version_tags_1)
		/sizeof(asn_DEF_Version_tags_1[0]), /* 1 */
	0,	/* No PER visible constraints */
	0, 0,	/* Defined elsewhere */
	0	/* No specifics */
};
```
Since the `asn_constr_check_f` is indirectly called to asn_DEF_INTEGER.check_constraints, so you do not encounter compilation error I had if `#define INTEGER_constraint asn_generic_no_constraint` not added.

So your code work fine is very reasonable, and my #143 work too because I just simplify these operation functions finding logic, so it is also reasonable. 

And the remain question are, 

(1) Why asn_DEF_INTEGER and asn_DEF_NativeInteger has asn_generic_no_constraint as their asn_constr_check_f fields ?

(2) Why the following ASN.1 statement does not generate constraint check function ?'
```
Version  ::=  INTEGER  {  v1(0), v2(1), v3(2)  }
```

Honestly, I don't know. 
And it is beyond my purpose of #143, I don't plan to tackle it here.